iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -222,28 +230,79 @@ TEST_F(ServiceDiscovery_test, FindServiceWithInstanceWildcardReturnsAllMatchingS
+    serviceContainerExp.push_back(SERVICE_DESCRIPTION1);
+    serviceContainerExp.push_back(SERVICE_DESCRIPTION3);
+
+    auto serviceContainer = sut.findService(iox::capro::Wildcard, INSTANCE, EVENT);
+    ASSERT_FALSE(serviceContainer.has_error());
+    ASSERT_THAT(serviceContainer.value().size(), Eq(2U));
+    EXPECT_TRUE(serviceContainer.value() == serviceContainerExp);
+}
+
+TEST_F(ServiceDiscovery_test, FindServiceWithEventWildcardReturnsCorrectServices)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e78b35f4-b3c3-4c39-b10a-67712c72e28a");
+    const IdString_t SERVICE = "service";
+    const IdString_t INSTANCE = "instance";
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION1(SERVICE, INSTANCE, "event1");
     iox::popo::UntypedPublisher publisher_sd1(SERVICE_DESCRIPTION1);
-    const iox::capro::ServiceDescription SERVICE_DESCRIPTION2("service2", "another_instance", "event2");
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION2("another_service", INSTANCE, "event2");
     iox::popo::UntypedPublisher publisher_sd2(SERVICE_DESCRIPTION2);
-    const iox::capro::ServiceDescription SERVICE_DESCRIPTION3("service3", INSTANCE, "event3");
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION3(SERVICE, INSTANCE, "event3");
     iox::popo::UntypedPublisher publisher_sd3(SERVICE_DESCRIPTION3);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION4(SERVICE, "another_instance", "event4");
+    iox::popo::UntypedPublisher publisher_sd4(SERVICE_DESCRIPTION4);
 
     ServiceContainer serviceContainerExp;
     serviceContainerExp.push_back(SERVICE_DESCRIPTION1);
     serviceContainerExp.push_back(SERVICE_DESCRIPTION3);
 
-    auto serviceContainer = sut.findService(iox::capro::Wildcard, INSTANCE);
+    auto serviceContainer = sut.findService(SERVICE, INSTANCE, iox::capro::Wildcard);
+    ASSERT_FALSE(serviceContainer.has_error());
+    ASSERT_THAT(serviceContainer.value().size(), Eq(2U));
+    EXPECT_TRUE(serviceContainer.value() == serviceContainerExp);
+}
+
+TEST_F(ServiceDiscovery_test, FindServiceWithInstanceWildcardReturnsCorrectServices)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2ec4b422-3ded-4af3-9e72-3b870c55031c");
+    const IdString_t SERVICE = "service";
+    const IdString_t EVENT = "event";
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION1(SERVICE, "instance1", EVENT);
+    iox::popo::UntypedPublisher publisher_sd1(SERVICE_DESCRIPTION1);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION2("another_service", "instance2", EVENT);
+    iox::popo::UntypedPublisher publisher_sd2(SERVICE_DESCRIPTION2);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION3(SERVICE, "instance3", EVENT);
+    iox::popo::UntypedPublisher publisher_sd3(SERVICE_DESCRIPTION3);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION4(SERVICE, "instance4", "another_event");
+    iox::popo::UntypedPublisher publisher_sd4(SERVICE_DESCRIPTION4);
```

Comment:
I understand the structure of the tests but somehow it looks like the naming of variables and strings makes it a little harder to follow then it needs to be.

Maybe we could use trailing numbers in the strings consistently and rename `publisher_sdx` into `publisher_x`.
I think this PR mainly used the existing scheme and this is fine.

The tests added make sense, but I have a hard time to say whether we are not missing something.

We could also create a new issue (stretch goal) to refactor `ServiceDiscovery` tests as needed (including analysis whether there are missing tests and adding missing tests).

Reply:
I would not invest any more time into those service discovery tests here. Leave them be and another poor soul has to refactor them completely sooner or later.

Reply:
Created #1065